src/java/org/apache/cassandra/audit/AuditLogOptionsCompositeData.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.audit;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.management.openmbean.CompositeData;
+import javax.management.openmbean.CompositeDataSupport;
+import javax.management.openmbean.CompositeType;
+import javax.management.openmbean.OpenDataException;
+import javax.management.openmbean.OpenType;
+import javax.management.openmbean.SimpleType;
+
+import org.apache.cassandra.config.ParameterizedClass;
+
+public class AuditLogOptionsCompositeData
```

Comment:
i'm thinking this class would be a better fit as an inner class to `GetAuditLog` since it is about a jmx represented view of the AuditLogOptions (i.e. let's keep the jmx specifics out of this package)

Reply:
While I understand what you mean, it looks weird to depend in StorageService on some stuff from the nodetool package.

Reply:
>  it looks weird to depend in StorageService on some stuff from the nodetool package.

Good point.

So two things here now come to mind…
1) why are we putting `getAuditLogOptions()` into StorageService? Would it be cleaner we a `AuditLogManagerMBean` ?
2) how do we code this to make it easy for the virtual table to be added?

WRT (2) the codebase is going to find new precedences to accommodate the transition period of JMX and Virtual Tables, and it would be nice if each didn't involve a verbose copy of code. (I'm not thinking about any high level compatibility layer as was discussed on the ML, just code with a small amount of effort that makes sense.)

Reply:
That is actually a good idea to create separate MBean for that. I might do that.

I really appreciate your willingness to help here in regards of upcoming changes related to virtual tables and similar, however, honestly, and I do not want to look like I am rushing this, but just the sake of the simplicity and getting rid of this, I would go with the easy path, not trying to come up with anything new, until we prepare CEP where all these things will be written down more clearly. I am afraid that anything we come up with here will be rewritten anyway when that virtual tables CEP arrives.

Reply:
There won't be a CEP for virtual tables, they already exist, and we are free to add virtual tables simply under improvement tickets. 

My suggestion and offer to write up a quick virtual table was not for inclusion in this patch (though it can be if want it), but for a quick evaluation of the technical debt possible avoided. Let me offer it up and I can then better explain.

Reply:
https://github.com/instaclustr/cassandra/compare/CASSANDRA-16725...thelastpickle:mck/instaclustr/CASSANDRA-16725/trunk